[Clang][Docs] Add release note for execute-only support on AArch64

Il-Capitano opened the pull request.

Il-Capitano committed:
[Clang][Docs] Add release note for execute-only support on AArch64

davemgreen reviewed (commented):
There is a `Arm and AArch64 Support` section too, is it worth adding a note there either instead or as well? It could maybe just be a note that points back to the Modified Compiler Flags section.

Il-Capitano committed:
Add note in "Arm and AArch64 support" section too

Il-Capitano commented:
> There is a `Arm and AArch64 Support` section too, is it worth adding a note there either instead or as well? It could maybe just be a note that points back to the Modified Compiler Flags section.

Thanks, I didn't notice that section.

I simplified the note in "Modified Compiler Flags" and added one in "Arm and AArch64 Support".

davemgreen reviewed (approved):
Thanks, LGTM
